subprojects/language-java/src/integTest/groovy/org/gradle/api/tasks/javadoc/JavadocToolchainIntegrationTest.groovy

```diff
@@ -20,10 +20,58 @@ import org.gradle.api.JavaVersion
 import org.gradle.integtests.fixtures.AbstractIntegrationSpec
 import org.gradle.integtests.fixtures.AvailableJavaHomes
 import org.gradle.internal.jvm.Jvm
+import org.gradle.internal.jvm.inspection.JvmInstallationMetadata
 import spock.lang.IgnoreIf
 
+import static org.gradle.integtests.fixtures.AvailableJavaHomes.getDifferentVersion
+import static org.gradle.integtests.fixtures.AvailableJavaHomes.getJvmInstallationMetadata
+
 class JavadocToolchainIntegrationTest extends AbstractIntegrationSpec {
 
+    def "changing toolchain invalidates task"() {
+        def jdkMetadata1 = getJvmInstallationMetadata(Jvm.current())
+        def jdkMetadata2 = getJvmInstallationMetadata(getDifferentVersion())
+
+        buildFile << """
+            plugins {
+                id 'java'
+            }
+
+            javadoc {
+                javadocTool = javaToolchains.javadocToolFor {
+                    def version = ${jdkMetadata1.languageVersion.majorVersion}
+                    version = providers.gradleProperty('test.javadoc.version').getOrElse(version)
+                    languageVersion = JavaLanguageVersion.of(version)
+                }
+            }
+        """
+        file('src/main/java/Lib.java') << testLib()
+
+        when:
+        withInstallations(jdkMetadata1, jdkMetadata2).run(":javadoc", "--info")
+        then:
+        executedAndNotSkipped(":javadoc")
+        file("build/docs/javadoc/Lib.html").text.contains("Some API documentation.")
+
+        when:
+        withInstallations(jdkMetadata1, jdkMetadata2).run(":javadoc", "--info")
+        then:
+        skipped(":javadoc")
+
+        when:
+        executer.withArgument("-Ptest.javadoc.version=${jdkMetadata2.languageVersion.majorVersion}")
+        withInstallations(jdkMetadata1, jdkMetadata2).run(":javadoc", "--info")
```

Comment:
Why run with `--info` in these invocations? You don't check the info log at all, right?

Reply:
Good catch. This may have been left there for debugging purposes, but can be removed.